src/contract_wrappers/modules/permission_manager/general_permission_manager_wrapper.ts

```diff
@@ -153,6 +153,7 @@ export default class GeneralPermissionManagerWrapper extends ModuleWrapper {
       delegate: params.delegate,
     });
     assert.assert(!delegate, 'Already present');
+    assert.assert(await this.isCallerAllowed(params.txData, 'CHANGE_PERMISSION'), 'Caller is not allowed');
```

Comment:
We can move this into assert class perhaps

Reply:
It would be great but I think if it's possible. It needs to make calls to contracts to check and it would make `assert` class very complex.